### PR TITLE
REFAPP-312: Swagger file request header in TPP Ref Server

### PR DESCRIPTION
**Pull request by mbana**
REFAPP-312: Swagger file request header in TPP Ref Server
https://openbanking.atlassian.net/browse/REFAPP-312

---
**`X-ACCOUNT-SWAGGERS`:**
* support additional swagger files.
* add tests for validating against `x-account-swaggers` headers.
* add basic and detail responses expected from transaction basic and detail calls to https://github.com/OpenBankingUK/tpp-reference-server/pull/168/files#diff-9df43f0632e0e71fa00115ff83e305beR1.
* use ES6 Map instead of plain object.
* increase timeouts in test due to fetch of api json.

**misc:**
* add `Makefile`: only supports running tests for now.
* add `winston` logging library. 
* `docker-compose.yml`: tidyup - use single quotes instead of double.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
     "superagent-bunyan": "^5.2.0",
     "swagger-tools": "^0.10.3",
     "url-pattern": "^1.0.3",
-    "uuid": "^3.2.1"
+    "uuid": "^3.2.1",
+    "winston": "^3.0.0"
```
Comment (robmckinnon) [inline package.json]: Corresponding package-lock needs to be committed. Adds an additional 27 packages. Do we really need a second logger?

File: app/session/request-headers.js [hunk 2]
```diff
@@ -2,13 +2,16 @@ const { fapiFinancialIdFor } = require('../authorisation-servers');
...
-const extractHeaders = async (headers) => {
+exports.extractHeaders = async (headers) => {
   const sessionId = headers['authorization'];
   const authorisationServerId = headers['x-authorization-server-id'];
   const fapiFinancialId = await fapiFinancialIdFor(authorisationServerId);
   const interactionId = headers['x-fapi-interaction-id'] || uuidv4();
   const username = await getUsername(sessionId);
   const validationRunId = headers['x-validation-run-id'];
+  // Set of swagger files to validate against
+  // https://openbanking.atlassian.net/browse/REFAPP-312
```
[the remark this answers is not in the document]
Comment (mbana) [reply app/session/request-headers.js]: done.

File: test/validator/fixtures/transactions.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
+exports.default = class Transactions {
```
[the remark this answers is not in the document]
Comment (mbana) [reply test/validator/fixtures/transactions.js]: done.

File: package.json (hunk 1 above)
Comment (mbana) [reply package.json]: done. thanks. good catch.

File: test/session/request-headers.test.js [hunk 4]
```diff
@@ -50,4 +51,19 @@ describe('extractHeaders from request headers', () => {
...
+
+  describe('when X-ACCOUNT-SWAGGERS in headers', () => {
+    it('returns headers with same X-ACCOUNT-SWAGGERS', async () => {
+      const interactionId = generatedInteractionId;
+      const accountSwaggers = [
+        'test_accountSwaggers_1',
+        'test_accountSwaggers_2',
+        'test_accountSwaggers_3',
+      ];
+      const headers = Object.assign({}, { 'x-account-swaggers': accountSwaggers }, requestHeaders);
```
[the remark this answers is not in the document]
Comment (mbana) [reply test/session/request-headers.test.js]: i'm looking into this.
thanks.

File: test/validator/validate-request-response.test.js [hunk 5]
```diff
@@ -84,4 +84,28 @@ describe('validate', () => {
...
+    // https://raw.githubusercontent.com/OpenBankingUK/account-info-api-spec/refapp-295-permission-specific-swagger-files/dist/v2.0.0/account-info-swagger.json
+    // https://raw.githubusercontent.com/OpenBankingUK/account-info-api-spec/refapp-295-permission-specific-swagger-files/dist/v2.0.0/account-info-swagger-basic.json
+    // https://raw.githubusercontent.com/OpenBankingUK/account-info-api-spec/refapp-295-permission-specific-swagger-files/dist/v2.0.0/account-info-swagger-detail.json
+    it('returns failedValidation false', async () => {
+      const detailsWithAccountSwaggers = Object.assign({}, details, {
+        accountSwaggers: [
+          'https://raw.githubusercontent.com/OpenBankingUK/account-info-api-spec/refapp-295-permission-specific-swagger-files/dist/v2.0.0/account-info-swagger.json',
+          'https://raw.githubusercontent.com/OpenBankingUK/account-info-api-spec/refapp-295-permission-specific-swagger-files/dist/v2.0.0/account-info-swagger-detail.json',
+        ],
+        scope: undefined,
```
[the remark this answers is not in the document]
Comment (mbana) [reply test/validator/validate-request-response.test.js]: i'm looking into this.
thanks.

File: app/utils/logger.js [hunk 6]
```diff
@@ -1,14 +1,17 @@
 const winston = require('winston');
 
+// Config:
 // https://github.com/winstonjs/winston/blob/master/examples/custom-timestamp.js
 // https://github.com/winstonjs/winston/issues/1135
+
+const level = process.env.NODE_ENV !== 'test' ? 'debug' : 'info';
```
[the remark this answers is not in the document]
Comment (mbana) [reply app/utils/logger.js]: `winston` is probably the most widely used logging library i thought we might standarise to this.

**Pull request merged**